### PR TITLE
Support encoding NSNull values in NSJsonSerialization. Fixes #937.

### LINKED ISSUE #937: NSJSONSerialization dataWithJSONObject does not support NSNull for values
We handle NSNull in isValidJSONObject, but not in actually creating the JSON object.

### PR DESCRIPTION
**Pull request by rajsesh-msft**

### CODE REVIEW

File: tests/unittests/Foundation/NSJSONSerializationTests.m [hunk 1]
```diff
@@ -117,11 +119,15 @@ void VerifyDataWithJSONObjectThrows(id testObject) {
     id testObject6 = @{ @"foo" : @"1", @"bar" : @"2" };
     id testObject7 = @{ @"foo" : @{ @"bar" : @"1" }, @"foo2" : @{ @"bar2" : @"2" } };
     id testObject8 = [NSUUID UUID];
+    id testObject9 = @{ @"key": @"value",
+                        @"nullkey":[NSNull null],
+                        @"array":@[@{@"subnullkey":[NSNull null]}]};
```
Comment (DHowett-MSFT) [inline tests/unittests/Foundation/NSJSONSerializationTests.m]: nit: please run this through `clang-format`

**Comment by DHowett-MSFT** [comment]
:shipit:

**Comment by DHowett-MSFT** [comment]
Are we capable of parsing a JSON `null` into **NSNull**?

**Comment by rajsesh-msft** [comment]
Yes. We didnt have the tests, which I have now added.

---
In reply to: [244250072](https://github.com/Microsoft/WinObjC/pull/953#issuecomment-244250072) [](ancestors = 244250072)

File: tests/unittests/Foundation/NSJSONSerializationTests.m (hunk 1 above)
Comment (rajsesh-msft) [inline tests/unittests/Foundation/NSJSONSerializationTests.m]: Long story.  Didn't have commit hook setup in my new enlistment.  Good catch.

---
In reply to: [77276409](https://github.com/Microsoft/WinObjC/pull/953#discussion_r77276409) [](ancestors = 77276409)

**Pull request merged**